packages/plugins/i18n/admin/src/middlewares/localePermission.ts

```diff
@@ -1,39 +1,51 @@
+import { Middleware } from '@reduxjs/toolkit';
+import { Permission } from '@strapi/helper-plugin';
 import get from 'lodash/get';
 
-const filterPermissionWithLocale = (locale: string) => (permission: any) =>
-  get(permission, 'properties.locales', []).indexOf(locale) !== -1;
-
-const localePermissionMiddleware = () => () => (next: any) => (action: any) => {
-  if (action.type !== 'ContentManager/RBACManager/SET_PERMISSIONS') {
-    return next(action);
-  }
+import { RootState } from '../store/reducers';
+
+/**
+ * TODO: is it possible to get the action types? How do we do it
+ * when actions are spread across multiple packages e.g. content-manager
+ * or content-type-builder?
+ */
```

Comment:
I believe this is kind of difficult, maybe keeping a registry of actions and letting other packages extend it.
But probably given the amount of time it would take it's not worth it to impleemnt